src/language-yaml/print/mapping-item.js

```diff
@@ -101,7 +101,8 @@ function printMappingItem(path, options, print) {
     hasLeadingComments(value.content) ||
     (hasEndComments(value) &&
       value.content &&
-      !isNode(value.content, ["mapping", "sequence"])) ||
+      !isNode(value.content, ["mapping", "sequence"]) &&
+      value.content.children?.length > 0) ||
```

Comment:
We need explicitly add logic for `flowMapping`, `flowSequence` not just checking `.children`

Reply:
Gotcha, I added the specific logic